pybedlite/overlap_detector.py

```diff
@@ -177,85 +180,35 @@ def from_bedrecord(cls: Type["Interval"], record: BedRecord) -> "Interval":
         )
 
 
-_GenericGenomicSpan = TypeVar(
-    "_GenericGenomicSpan",
-    bound=Union[_GenomicSpanWithChrom, _GenomicSpanWithContig, _GenomicSpanWithRefName],
-)
+GenericGenomicsSpan = TypeVar("GenericGenomicsSpan", bound=Union[GenomicSpan, StrandedGenomicSpan])
```

Comment:
```suggestion
GenericGenomicSpan = TypeVar("GenericGenomicSpan", bound=Union[GenomicSpan, StrandedGenomicSpan]) 
```

This should be named consistently with the protocols.